### DISCUSSION TITLE
Error while loading transcriptome_fasta for Kallisto Index

### DISCUSSION THREAD

**Discussion by santataRU** [GuyTeichman/RNAlysis#39] [General]
Hi, All,

I encountered an error while loading the transcriptome_fasta for the Kallisto Index. I have attached the image and error messages below.

I would appreciate any advice on how to fix this error.

Thanks,

Xiao

<img width="820" alt="Screenshot 2024-07-22 at 5 55 17 PM" src="https://github.com/user-attachments/assets/e9d777d7-363d-434a-ae81-36696731859f">

Traceback (most recent call last):
  File "rnalysis/utils/io.py", line 2451, in generate_base_call
  File "rnalysis/utils/io.py", line 2392, in run_subprocess
  File "subprocess.py", line 1026, in __init__
  File "subprocess.py", line 1955, in _execute_child
FileNotFoundError: [Errno 2] No such file or directory: 'kallisto'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "rnalysis/gui/gui.py", line 4442, in finish_generic_job
  File "rnalysis/gui/gui_widgets.py", line 387, in run
  File "rnalysis/fastq.py", line 1645, in kallisto_create_index
  File "rnalysis/utils/io.py", line 2455, in generate_base_call
FileNotFoundError: RNAlysis could not find 'kallisto'. Please ensure that your installation folder is correct, or add it to PATH.

**Comment by GuyTeichman** [comment]
Hi Xiao,
I'm guessing that kallisto is not installed on your computer (it isn't included with RNAlysis installation). You can download kallisto using brew:
`ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`
`brew install kallisto`

Or follow the instructions here:
https://pachterlab.github.io/kallisto/download

Please let me know if that worked out for you. 

Best,
Guy.

**Reply by santataRU** [reply]
Thanks. I installed Kallisto, but received the following error while in the Kallisto build index: 

(base) bieniaszlab@Pauls-iMac-2 ~ % /Applications/RNAlysis ; exit;
Matplotlib is building the font cache; this may take a moment.
Warning: Failed to retreive legal taxons from PantherDB. Some features may not work as intended. To fix this issue, make sure your computer has internet connection, and restart RNAlysis. 
Warning: Failed to retreive legal taxons from Ensembl. Some features may not work as intended. To fix this issue, make sure your computer has internet connection, and restart RNAlysis. 
Warning: coroutine 'download_video' was never awaited
qt.qpa.fonts: Populating font family aliases took 89 ms. Replace uses of missing font family "New Roman" with one that exists to avoid this cost. 
Building kallisto index:   0%|                         | 0/1 [00:00<?, ?index/s]
Matplotlib is building the font cache; this may take a moment.
Warning: Failed to retreive legal taxons from PantherDB. Some features may not work as intended. To fix this issue, make sure your computer has internet connection, and restart RNAlysis. 
Warning: Failed to retreive legal taxons from Ensembl. Some features may not work as intended. To fix this issue, make sure your computer has internet connection, and restart RNAlysis.
